src/tasks/rebuild-web-api.js

```diff
@@ -120,10 +122,11 @@ module.exports = function rebuildWebApi(functionName, functionVersion, restApiId
 					}));
 				},
 				authorizerId = function () {
-					return methodOptions && methodOptions.customAuthorizer && authorizerIds[methodOptions.customAuthorizer];
+					const authorizerName = methodOptions.customAuthorizer || methodOptions.cognitoAuthorizer;
+					return methodOptions && authorizerName && authorizerIds[authorizerName];
 				},
 				parameters = flattenRequestParameters(methodOptions.requestParameters, path);
-			return apiGateway.putMethodPromise({
+			return apiGateway.putMethodAsync({
```

Comment:
why change this? xxxPromise methods come from the logging wrapper so we get console logs on api gateway calls.